File: sources/net.sf.j2s.core/test/dev/j2s/org/sun/awt/PostEventQueue.js
Clazz.declarePackage ("sun.awt");
Clazz.load (null, "sun.awt.PostEventQueue", ["sun.awt.EventQueueItem", "$.SunToolkit", "swingjs.JSToolkit"], function () {
c$ = Clazz.decorateAsClass (function () {
this.queueHead = null;
this.queueTail = null;
this.eventQueue = null;
Clazz.instantialize (this, arguments);
}, sun.awt, "PostEventQueue");
Clazz.makeConstructor (c$, 
function (eq) {
this.eventQueue = eq;
}, "java.awt.EventQueue");
Clazz.defineMethod (c$, "noEvents", 
function () {
return this.queueHead == null;
});
Clazz.defineMethod (c$, "flush", 
function () {
if (this.queueHead != null) {
var tempQueue;
{
tempQueue = this.queueHead;
this.queueHead = this.queueTail = null;
while (tempQueue != null) {
swingjs.JSToolkit.alert ("postevent IS NOT IMPLEMENTED " + tempQueue.event);
this.eventQueue.postEvent (tempQueue.event);
tempQueue = tempQueue.next;
}
}}});
Clazz.defineMethod (c$, "postEvent", 
function (event) {
var item =  new sun.awt.EventQueueItem (event);
{
if (this.queueHead == null) {
this.queueHead = this.queueTail = item;
} else {
this.queueTail.next = item;
this.queueTail = item;
}}sun.awt.SunToolkit.wakeupEventQueue (this.eventQueue, false);
}, "java.awt.AWTEvent");
});

File: sources/net.sf.j2s.java.core/src/sun/awt/PostEventQueue.java
package sun.awt;

import java.awt.AWTEvent;
import java.awt.EventQueue;

/*
 * PostEventQueue is a Thread that runs in the same AppContext as the
 * Java EventQueue.  It is a queue of AWTEvents to be posted to the
 * Java EventQueue.  The toolkit Thread (AWT-Windows/AWT-Motif) posts
 * events to this queue, which then calls EventQueue.postEvent().
 *
 * We do this because EventQueue.postEvent() may be overridden by client
 * code, and we mustn't ever call client code from the toolkit thread.
 */
public class PostEventQueue {
    private EventQueueItem queueHead = null;
    private EventQueueItem queueTail = null;
    private final EventQueue eventQueue;

    public PostEventQueue(EventQueue eq) {
        eventQueue = eq;
    }

    public boolean noEvents() {
        return queueHead == null;
    }

    /*
     * Continually post pending AWTEvents to the Java EventQueue.
     */
    public void flush() {
        if (queueHead != null) {
            EventQueueItem tempQueue;
            /*
             * We have to execute the loop inside the synchronized block
             * to ensure that the flush is completed before a new event
             * can be posted to this queue.
             */
            synchronized (this) {
                tempQueue = queueHead;
                queueHead = queueTail = null;
                /*
                 * If this PostEventQueue is flushed in parallel on two
                 * different threads tempQueue will be null for one of them.
                 */
                while (tempQueue != null) {
//                	 System.out.println("PostEventQueue posting tempEvent " + tempQueue.event);
                    eventQueue.postEvent(tempQueue.event);
                    tempQueue = tempQueue.next;
                }
            }
        }
    }

    /*
     * Enqueue an AWTEvent to be posted to the Java EventQueue.
     */
    void postEvent(AWTEvent event) {
        EventQueueItem item = new EventQueueItem(event);

        synchronized (this) {
            if (queueHead == null) {
                queueHead = queueTail = item;
            } else {
                queueTail.next = item;
                queueTail = item;
            }
        }
      SunToolkit.wakeupEventQueue(eventQueue, false);//SwingJS was event.getSource() == AWTAutoShutdown.getInstance());
    }
} 
// SwingJS  -- hidden classes EventQueueItem and were removed from this file